parquet-cli/src/main/java/org/apache/parquet/cli/commands/ShowFooterCommand.java

```diff
@@ -0,0 +1,150 @@
+import org.apache.parquet.hadoop.util.HadoopInputFile;
+import org.apache.parquet.io.InputFile;
+import org.apache.parquet.io.SeekableInputStream;
+import org.slf4j.Logger;
+
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+@Parameters(commandDescription = "Print the Parquet file footer in json format")
+public class ShowFooterCommand extends BaseCommand {
+
+  public ShowFooterCommand(Logger console) {
+    super(console);
+  }
+
+  @Parameter(description = "<parquet path>", required = true)
+  String target;
+
+  @Parameter(names = { "-r", "--raw" }, description = "Print the raw thrift object of the footer")
+  boolean raw = false;
+
+  @Override
+  public int run() throws IOException {
+    InputFile inputFile = HadoopInputFile.fromPath(qualifiedPath(target), getConf());
+
+    String json;
+    if (raw) {
+      json = readRawFooter(inputFile);
+    } else {
+      json = readFooter(inputFile);
+    }
+    console.info(json);
+
+    return 0;
+  }
+
+  private String readFooter(InputFile inputFile) throws JsonProcessingException, IOException {
+    String json;
+    try (ParquetFileReader reader = ParquetFileReader.open(inputFile)) {
+      ParquetMetadata footer = reader.getFooter();
+      ObjectMapper mapper = createObjectMapper();
+      mapper.setVisibility(PropertyAccessor.ALL, Visibility.NONE);
+      mapper.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+      json = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(footer);
+    }
+    return json;
+  }
+
+  private ObjectMapper createObjectMapper() {
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
+    mapper.configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, true);
+    return mapper;
+  }
+
+  private String readRawFooter(InputFile file) throws IOException {
+    long fileLen = file.getLength();
+
+    int FOOTER_LENGTH_SIZE = 4;
+    if (fileLen < MAGIC.length + FOOTER_LENGTH_SIZE + MAGIC.length) { // MAGIC + data + footer + footerIndex + MAGIC
+      throw new RuntimeException("Not a Parquet file (length is too low: " + fileLen + ")");
+    }
+
+    try (SeekableInputStream f = file.newStream()) {
+      // Read footer length and magic string - with a single seek
+      byte[] magic = new byte[MAGIC.length];
+      long fileMetadataLengthIndex = fileLen - magic.length - FOOTER_LENGTH_SIZE;
+      f.seek(fileMetadataLengthIndex);
+      int fileMetadataLength = readIntLittleEndian(f);
+      f.readFully(magic);
+
+      if (Arrays.equals(EFMAGIC, magic)) {
+        throw new RuntimeException("Encrypted Parquet files are not supported.");
```

Comment:
technically, EFMAGIC is for encrypted files with encrypted footer. Encrypted files can also have the regular MAGIC, if created in plaintext footer mode.

Reply:
Thanks, @ggershinsky for the correction. Updated the message.

Reply:
@gszadovszky Did you try the plaintext footer encrypted file to see if this change work?

Reply:
@shangxinli, just checked it offline. It works fine. (The unit test I've added lately does not contain encrypted file because it would require much more efforts and I think it does not worth it.)